autoprotocol/builders.py

```diff
@@ -1531,9 +1531,15 @@ def mode_params(
 
     @staticmethod
     def device_mode_params(
+        device: str,
         model: Optional[str] = None,
         chip_material: Optional[str] = None,
         nozzle: Optional[str] = None,
+        diaphragm: Optional[int] = None,
+        nozzle_size: Optional[Unit] or Optional[str] = None,
```

Comment:
you want the `Union` type

```suggestion
        nozzle_size: Optional[Union[Unit, str]] = None,
```